File: python/51-60/Merge Intervals.py
 # @param intervals, a list of Interval
    # @return a list of Interval
    def merge(self, intervals):
        if not intervals or len(intervals)<=1:
            return intervals
        B=sorted(intervals,key=lambda x:(x.start,x.end))
        result = []
        cur = Interval(B[0].start,B[0].end)
        i = 1
        while i <len(B):
            if cur.end <B[i].start:
                result.append(cur)
                cur=B[i]
            elif cur.end <B[i].end:
                cur.end=B[i].end
            i+=1
        result.append(cur)
        return result

File: Cpp/51-60/Merge Intervals.cpp
/**
 * Definition for an interval.
 * struct Interval {
 *     int start;
 *     int end;
 *     Interval() : start(0), end(0) {}
 *     Interval(int s, int e) : start(s), end(e) {}
 * };
 */
 
 /*
  
  run time: 584ms;
  error: 1   cmp放在class内了
 
 */
bool cmp(const Interval& a, const Interval& b){
      return a.start<b.start;
    }
class Solution {
public:
    vector<Interval> merge(vector<Interval>& intervals) {
        vector<Interval> res;
        if(intervals.size()==0) return res;
        sort(intervals.begin(),intervals.end(),cmp);
        Interval temp = intervals[0];
        for(int i=1; i<intervals.size(); i++){
          if(temp.end<intervals[i].start){
            res.push_back(temp);
            temp = intervals[i];
          }
          else temp.end = max(temp.end,intervals[i].end);
        }
        res.push_back(temp);
    }
};
